### ISSUE TITLE
[forge] 1.20.4 

### ISSUE THREAD

**Issue by vrozkovec** [noriokun4649/TorchBowMod#17]
Hi, if it is not too much work, would it be possible, please, to make this mod compatible with the most recent v1.20.4 version and forge mod loader? Thank you.

**Comment by vrozkovec** [comment]
Here's a crash log:

```
---- Minecraft Crash Report ----
// Daisy, daisy...

Time: 2024-01-04 16:46:37
Description: Mod loading error has occurred

java.lang.Exception: Mod Loading has failed
	at net.minecraftforge.logging.CrashReportExtender.dumpModLoadingCrashReport(CrashReportExtender.java:48) ~[forge-1.20.4-49.0.14-universal.jar:?] {re:classloading}
	at net.minecraftforge.client.loading.ClientModLoader.completeModLoading(ClientModLoader.java:119) ~[forge-1.20.4-49.0.14-universal.jar:?] {re:classloading,pl:runtimedistcleaner:A}
	at net.minecraft.client.Minecraft.lambda$new$6(Minecraft.java:611) ~[forge-1.20.4-49.0.14-client.jar:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.Util.m_137521_(Util.java:432) ~[forge-1.20.4-49.0.14-client.jar:?] {re:classloading}
	at net.minecraft.client.Minecraft.lambda$new$7(Minecraft.java:602) ~[forge-1.20.4-49.0.14-client.jar:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraftforge.client.loading.ForgeLoadingOverlay.m_88315_(ForgeLoadingOverlay.java:146) ~[forge-1.20.4-49.0.14-universal.jar:?] {re:classloading}
	at net.minecraft.client.renderer.GameRenderer.m_109093_(GameRenderer.java:931) ~[forge-1.20.4-49.0.14-client.jar:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.client.Minecraft.m_91383_(Minecraft.java:1250) ~[forge-1.20.4-49.0.14-client.jar:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.client.Minecraft.m_91374_(Minecraft.java:801) ~[forge-1.20.4-49.0.14-client.jar:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.client.main.Main.main(Main.java:234) ~[forge-1.20.4-49.0.14-client.jar:?] {re:classloading,pl:runtimedistcleaner:A}
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?] {}
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?] {}
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?] {}
	at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?] {}
	at net.minecraftforge.fml.loading.targets.CommonLaunchHandler.runTarget(CommonLaunchHandler.java:91) ~[fmlloader-1.20.4-49.0.14.jar:?] {}
	at net.minecraftforge.fml.loading.targets.CommonLaunchHandler.lambda$makeService$0(CommonLaunchHandler.java:75) ~[fmlloader-1.20.4-49.0.14.jar:?] {}
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:54) ~[modlauncher-10.1.2.jar:?] {}
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:74) ~[modlauncher-10.1.2.jar:?] {}
	at cpw.mods.modlauncher.Launcher.run(Launcher.java:114) ~[modlauncher-10.1.2.jar:?] {}
	at cpw.mods.modlauncher.Launcher.main(Launcher.java:73) ~[modlauncher-10.1.2.jar:?] {}
	at cpw.mods.modlauncher.BootstrapEntry.main(BootstrapEntry.java:17) ~[modlauncher-10.1.2.jar:?] {}
	at net.minecraftforge.bootstrap.Bootstrap.moduleMain(Bootstrap.java:97) ~[bootstrap-2.0.0.jar!/:?] {}
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?] {}
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?] {}
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?] {}
	at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?] {}
	at net.minecraftforge.bootstrap.Bootstrap.start(Bootstrap.java:48) ~[bootstrap-2.0.0.jar!/:?] {}
	at net.minecraftforge.bootstrap.ForgeBootstrap.main(ForgeBootstrap.java:18) ~[bootstrap-2.0.0.jar!/:?] {}


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Render thread
Suspected Mods: NONE
Stacktrace:
	at java.net.URI$Parser.fail(URI.java:2976) ~[?:?] {}
-- MOD torchbowmod --
Details:
	Caused by 0: java.lang.IllegalArgumentException: Illegal character in path at index 33: file:///home/vit/.minecraft/mods/[1.20][Forge]TorchBowMod_v1.0.jar
		at jdk.nio.zipfs.ZipFileSystemProvider.uriToPath(ZipFileSystemProvider.java:78) ~[jdk.zipfs:?] {}
		at jdk.nio.zipfs.ZipFileSystemProvider.getFileSystem(ZipFileSystemProvider.java:151) ~[jdk.zipfs:?] {}
		at jdk.nio.zipfs.ZipFileSystemProvider.getPath(ZipFileSystemProvider.java:142) ~[jdk.zipfs:?] {}
		at java.nio.file.Path.of(Path.java:208) ~[?:?] {}
		at java.nio.file.Paths.get(Paths.java:98) ~[?:?] {}
		at java.util.Optional.map(Optional.java:260) ~[?:?] {}
		at cpw.mods.jarhandling.impl.Jar$JarModuleDataProvider.open(Jar.java:416) ~[securemodules-2.2.9.jar:2.2.9] {}
		at net.minecraftforge.securemodules.SecureModuleFinder$Reader.open(SecureModuleFinder.java:97) ~[securemodules-2.2.9.jar:2.2.9] {}
		at net.minecraftforge.securemodules.SecureModuleClassLoader.getClassBytes(SecureModuleClassLoader.java:170) ~[securemodules-2.2.9.jar:2.2.9] {}
		at cpw.mods.cl.ModuleClassLoader.getClassBytes(ModuleClassLoader.java:38) ~[securemodules-2.2.9.jar:2.2.9] {}
		at net.minecraftforge.securemodules.SecureModuleClassLoader.readerToClass(SecureModuleClassLoader.java:468) ~[securemodules-2.2.9.jar:2.2.9] {}
		at net.minecraftforge.securemodules.SecureModuleClassLoader.findClass(SecureModuleClassLoader.java:399) ~[securemodules-2.2.9.jar:2.2.9] {}
		at java.lang.ClassLoader.loadClass(ClassLoader.java:637) ~[?:?] {}
		at java.lang.Class.forName(Class.java:545) ~[?:?] {}
		at net.minecraftforge.fml.javafmlmod.FMLModContainer.<init>(FMLModContainer.java:51) ~[javafmllanguage-1.20.4-49.0.14.jar:49.0.14] {}
		at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:?] {}
		at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77) ~[?:?] {}
		at jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[?:?] {}
		at java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499) ~[?:?] {}
		at java.lang.reflect.Constructor.newInstance(Constructor.java:480) ~[?:?] {}
		at net.minecraftforge.fml.javafmlmod.FMLJavaModLanguageProvider$FMLModTarget.loadMod(FMLJavaModLanguageProvider.java:62) ~[javafmllanguage-1.20.4-49.0.14.jar:49.0.14] {}
		at net.minecraftforge.fml.ModLoader.buildModContainerFromTOML(ModLoader.java:291) ~[fmlcore-1.20.4-49.0.14.jar:1.0] {}
		at net.minecraftforge.fml.ModLoader.lambda$buildMods$24(ModLoader.java:258) ~[fmlcore-1.20.4-49.0.14.jar:1.0] {}
		at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197) ~[?:?] {}
		at java.util.HashMap$EntrySpliterator.forEachRemaining(HashMap.java:1850) ~[?:?] {}
		at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509) ~[?:?] {}
		at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499) ~[?:?] {}
		at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921) ~[?:?] {}
		at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:?] {}
		at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682) ~[?:?] {}
		at net.minecraftforge.fml.ModLoader.buildMods(ModLoader.java:260) ~[fmlcore-1.20.4-49.0.14.jar:1.0] {}
		at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197) ~[?:?] {}
		at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197) ~[?:?] {}
		at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1625) ~[?:?] {}
		at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509) ~[?:?] {}
		at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499) ~[?:?] {}
		at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:575) ~[?:?] {}
		at java.util.stream.AbstractPipeline.evaluateToArrayNode(AbstractPipeline.java:260) ~[?:?] {}
		at java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:616) ~[?:?] {}
		at java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:622) ~[?:?] {}
		at java.util.stream.ReferencePipeline.toList(ReferencePipeline.java:627) ~[?:?] {}
		at net.minecraftforge.fml.ModLoader.gatherAndInitializeMods(ModLoader.java:171) ~[fmlcore-1.20.4-49.0.14.jar:1.0] {}
		at net.minecraftforge.client.loading.ClientModLoader.lambda$begin$1(ClientModLoader.java:52) ~[forge-1.20.4-49.0.14-universal.jar:?] {re:classloading,pl:runtimedistcleaner:A}
		at net.minecraftforge.client.loading.ClientModLoader.lambda$createRunnableWithCatch$4(ClientModLoader.java:72) ~[forge-1.20.4-49.0.14-universal.jar:?] {re:classloading,pl:runtimedistcleaner:A}
		at net.minecraftforge.client.loading.ClientModLoader.begin(ClientModLoader.java:52) ~[forge-1.20.4-49.0.14-universal.jar:?] {re:classloading,pl:runtimedistcleaner:A}
		at net.minecraft.client.Minecraft.<init>(Minecraft.java:484) ~[forge-1.20.4-49.0.14-client.jar:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A}
		at net.minecraft.client.main.Main.main(Main.java:196) ~[forge-1.20.4-49.0.14-client.jar:?] {re:classloading,pl:runtimedistcleaner:A}
		at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?] {}
		at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?] {}
		at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?] {}
		at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?] {}
		at net.minecraftforge.fml.loading.targets.CommonLaunchHandler.runTarget(CommonLaunchHandler.java:91) ~[fmlloader-1.20.4-49.0.14.jar:?] {}
		at net.minecraftforge.fml.loading.targets.CommonLaunchHandler.lambda$makeService$0(CommonLaunchHandler.java:75) ~[fmlloader-1.20.4-49.0.14.jar:?] {}
		at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:54) ~[modlauncher-10.1.2.jar:?] {}
		at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:74) ~[modlauncher-10.1.2.jar:?] {}
		at cpw.mods.modlauncher.Launcher.run(Launcher.java:114) ~[modlauncher-10.1.2.jar:?] {}
		at cpw.mods.modlauncher.Launcher.main(Launcher.java:73) ~[modlauncher-10.1.2.jar:?] {}
		at cpw.mods.modlauncher.BootstrapEntry.main(BootstrapEntry.java:17) ~[modlauncher-10.1.2.jar:?] {}
		at net.minecraftforge.bootstrap.Bootstrap.moduleMain(Bootstrap.java:97) ~[bootstrap-2.0.0.jar!/:?] {}
		at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?] {}
		at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?] {}
		at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?] {}
		at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?] {}
		at net.minecraftforge.bootstrap.Bootstrap.start(Bootstrap.java:48) ~[bootstrap-2.0.0.jar!/:?] {}
		at net.minecraftforge.bootstrap.ForgeBootstrap.main(ForgeBootstrap.java:18) ~[bootstrap-2.0.0.jar!/:?] {}

	Mod File: /home/vit/.minecraft/mods/[1.20][Forge]TorchBowMod_v1.0.jar
	Failure message: TorchBowMod has class loading errors
		java.lang.IllegalArgumentException: Illegal character in path at index 33: file:///home/vit/.minecraft/mods/[1.20][Forge]TorchBowMod_v1.0.jar
	Mod Version: 1.0.0
	Mod Issue URL: NOT PROVIDED
	Exception message: java.net.URISyntaxException: Illegal character in path at index 33: file:///home/vit/.minecraft/mods/[1.20][Forge]TorchBowMod_v1.0.jar
Stacktrace:
	at java.net.URI$Parser.fail(URI.java:2976) ~[?:?] {}
	at java.net.URI$Parser.checkChars(URI.java:3147) ~[?:?] {}
	at java.net.URI$Parser.parseHierarchical(URI.java:3229) ~[?:?] {}
	at java.net.URI$Parser.parse(URI.java:3177) ~[?:?] {}
	at java.net.URI.<init>(URI.java:623) ~[?:?] {}
	at jdk.nio.zipfs.ZipFileSystemProvider.uriToPath(ZipFileSystemProvider.java:76) ~[jdk.zipfs:?] {}
	at jdk.nio.zipfs.ZipFileSystemProvider.getFileSystem(ZipFileSystemProvider.java:151) ~[jdk.zipfs:?] {}
	at jdk.nio.zipfs.ZipFileSystemProvider.getPath(ZipFileSystemProvider.java:142) ~[jdk.zipfs:?] {}
	at java.nio.file.Path.of(Path.java:208) ~[?:?] {}
	at java.nio.file.Paths.get(Paths.java:98) ~[?:?] {}
	at java.util.Optional.map(Optional.java:260) ~[?:?] {}
	at cpw.mods.jarhandling.impl.Jar$JarModuleDataProvider.open(Jar.java:416) ~[securemodules-2.2.9.jar:2.2.9] {}
	at net.minecraftforge.securemodules.SecureModuleFinder$Reader.open(SecureModuleFinder.java:97) ~[securemodules-2.2.9.jar:2.2.9] {}
	at net.minecraftforge.securemodules.SecureModuleClassLoader.getClassBytes(SecureModuleClassLoader.java:170) ~[securemodules-2.2.9.jar:2.2.9] {}
	at cpw.mods.cl.ModuleClassLoader.getClassBytes(ModuleClassLoader.java:38) ~[securemodules-2.2.9.jar:2.2.9] {}
	at net.minecraftforge.securemodules.SecureModuleClassLoader.readerToClass(SecureModuleClassLoader.java:468) ~[securemodules-2.2.9.jar:2.2.9] {}
	at net.minecraftforge.securemodules.SecureModuleClassLoader.findClass(SecureModuleClassLoader.java:399) ~[securemodules-2.2.9.jar:2.2.9] {}
	at java.lang.ClassLoader.loadClass(ClassLoader.java:637) ~[?:?] {}
	at java.lang.Class.forName(Class.java:545) ~[?:?] {}
	at net.minecraftforge.fml.javafmlmod.FMLModContainer.<init>(FMLModContainer.java:51) ~[javafmllanguage-1.20.4-49.0.14.jar:49.0.14] {}
	at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:?] {}
	at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77) ~[?:?] {}
	at jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[?:?] {}
	at java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499) ~[?:?] {}
	at java.lang.reflect.Constructor.newInstance(Constructor.java:480) ~[?:?] {}
	at net.minecraftforge.fml.javafmlmod.FMLJavaModLanguageProvider$FMLModTarget.loadMod(FMLJavaModLanguageProvider.java:62) ~[javafmllanguage-1.20.4-49.0.14.jar:49.0.14] {}
	at net.minecraftforge.fml.ModLoader.buildModContainerFromTOML(ModLoader.java:291) ~[fmlcore-1.20.4-49.0.14.jar:1.0] {}
	at net.minecraftforge.fml.ModLoader.lambda$buildMods$24(ModLoader.java:258) ~[fmlcore-1.20.4-49.0.14.jar:1.0] {}
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197) ~[?:?] {}
	at java.util.HashMap$EntrySpliterator.forEachRemaining(HashMap.java:1850) ~[?:?] {}
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509) ~[?:?] {}
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499) ~[?:?] {}
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921) ~[?:?] {}
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:?] {}
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682) ~[?:?] {}
	at net.minecraftforge.fml.ModLoader.buildMods(ModLoader.java:260) ~[fmlcore-1.20.4-49.0.14.jar:1.0] {}
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197) ~[?:?] {}
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197) ~[?:?] {}
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1625) ~[?:?] {}
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509) ~[?:?] {}
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499) ~[?:?] {}
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:575) ~[?:?] {}
	at java.util.stream.AbstractPipeline.evaluateToArrayNode(AbstractPipeline.java:260) ~[?:?] {}
	at java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:616) ~[?:?] {}
	at java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:622) ~[?:?] {}
	at java.util.stream.ReferencePipeline.toList(ReferencePipeline.java:627) ~[?:?] {}
	at net.minecraftforge.fml.ModLoader.gatherAndInitializeMods(ModLoader.java:171) ~[fmlcore-1.20.4-49.0.14.jar:1.0] {}
	at net.minecraftforge.client.loading.ClientModLoader.lambda$begin$1(ClientModLoader.java:52) ~[forge-1.20.4-49.0.14-universal.jar:?] {re:classloading,pl:runtimedistcleaner:A}
	at net.minecraftforge.client.loading.ClientModLoader.lambda$createRunnableWithCatch$4(ClientModLoader.java:72) ~[forge-1.20.4-49.0.14-universal.jar:?] {re:classloading,pl:runtimedistcleaner:A}
	at net.minecraftforge.client.loading.ClientModLoader.begin(ClientModLoader.java:52) ~[forge-1.20.4-49.0.14-universal.jar:?] {re:classloading,pl:runtimedistcleaner:A}
	at net.minecraft.client.Minecraft.<init>(Minecraft.java:484) ~[forge-1.20.4-49.0.14-client.jar:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.client.main.Main.main(Main.java:196) ~[forge-1.20.4-49.0.14-client.jar:?] {re:classloading,pl:runtimedistcleaner:A}
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?] {}
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?] {}
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?] {}
	at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?] {}
	at net.minecraftforge.fml.loading.targets.CommonLaunchHandler.runTarget(CommonLaunchHandler.java:91) ~[fmlloader-1.20.4-49.0.14.jar:?] {}
	at net.minecraftforge.fml.loading.targets.CommonLaunchHandler.lambda$makeService$0(CommonLaunchHandler.java:75) ~[fmlloader-1.20.4-49.0.14.jar:?] {}
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:54) ~[modlauncher-10.1.2.jar:?] {}
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:74) ~[modlauncher-10.1.2.jar:?] {}
	at cpw.mods.modlauncher.Launcher.run(Launcher.java:114) ~[modlauncher-10.1.2.jar:?] {}
	at cpw.mods.modlauncher.Launcher.main(Launcher.java:73) ~[modlauncher-10.1.2.jar:?] {}
	at cpw.mods.modlauncher.BootstrapEntry.main(BootstrapEntry.java:17) ~[modlauncher-10.1.2.jar:?] {}
	at net.minecraftforge.bootstrap.Bootstrap.moduleMain(Bootstrap.java:97) ~[bootstrap-2.0.0.jar!/:?] {}
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?] {}
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?] {}
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?] {}
	at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?] {}
	at net.minecraftforge.bootstrap.Bootstrap.start(Bootstrap.java:48) ~[bootstrap-2.0.0.jar!/:?] {}
	at net.minecraftforge.bootstrap.ForgeBootstrap.main(ForgeBootstrap.java:18) ~[bootstrap-2.0.0.jar!/:?] {}


-- System Details --
Details:
	Minecraft Version: 1.20.4
	Minecraft Version ID: 1.20.4
	Operating System: Linux (amd64) version 6.2.0-39-generic
	Java Version: 17.0.8, Microsoft
	Java VM Version: OpenJDK 64-Bit Server VM (mixed mode), Microsoft
	Memory: 400057536 bytes (381 MiB) / 738197504 bytes (704 MiB) up to 2147483648 bytes (2048 MiB)
	CPUs: 8
	Processor Vendor: GenuineIntel
	Processor Name: Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz
	Identifier: Intel64 Family 6 Model 158 Stepping 9
	Microarchitecture: Kaby Lake
	Frequency (GHz): 2.80
	Number of physical packages: 1
	Number of physical CPUs: 4
	Number of logical CPUs: 8
	Graphics card #0 name: HD Graphics 630
	Graphics card #0 vendor: Intel Corporation (0x8086)
	Graphics card #0 VRAM (MB): 256.00
	Graphics card #0 deviceId: 0x591b
	Graphics card #0 versionInfo: unknown
	Graphics card #1 name: GP107M [GeForce GTX 1050 Ti Mobile]
	Graphics card #1 vendor: NVIDIA Corporation (0x10de)
	Graphics card #1 VRAM (MB): 288.00
	Graphics card #1 deviceId: 0x1c8c
	Graphics card #1 versionInfo: unknown
	Virtual memory max (MB): 31608.59
	Virtual memory used (MB): 15708.86
	Swap memory total (MB): 15625.00
	Swap memory used (MB): 30.00
	JVM Flags: 8 total; -Xss1M -Xmx2G -XX:+UnlockExperimentalVMOptions -XX:+UseG1GC -XX:G1NewSizePercent=20 -XX:G1ReservePercent=20 -XX:MaxGCPauseMillis=50 -XX:G1HeapRegionSize=32M
	ModLauncher: 10.1.2
	ModLauncher launch target: forge_client
	ModLauncher naming: srg
	ModLauncher services: 
		/ slf4jfixer PLUGINSERVICE 
		/ runtimedistcleaner PLUGINSERVICE 
		/ runtime_enum_extender PLUGINSERVICE 
		/ object_holder_definalize PLUGINSERVICE 
		/ capability_token_subclass PLUGINSERVICE 
		/ accesstransformer PLUGINSERVICE 
		/ eventbus PLUGINSERVICE 
		/ mixin PLUGINSERVICE 
		/ fml TRANSFORMATIONSERVICE 
		/ mixin TRANSFORMATIONSERVICE 
	FML Language Providers: 
		lowcodefml@49
		minecraft@1.0
		javafml@49.0.14
	Mod List: 
		forge-1.20.4-49.0.14-client.jar                   |Minecraft                     |minecraft                     |1.20.4              |NONE      |Manifest: NOSIGNATURE
		dynamiclights-1.20.4.1.jar                        |Dynamic Lights                |dynamiclights                 |1.20.4.1            |NONE      |Manifest: NOSIGNATURE
		forge-1.20.4-49.0.14-universal.jar                |Forge                         |forge                         |49.0.14             |NONE      |Manifest: NOSIGNATURE
		[1.20][Forge]TorchBowMod_v1.0.jar                 |TorchBowMod                   |torchbowmod                   |1.0.0               |NONE      |Manifest: NOSIGNATURE
```

**Comment by noriokun4649** [comment]
It is possible to support Forge 1.20.4, but Forge 1.20.4 has not yet been released in stable form. As a result, it is often pointless due to frequent specification changes. Therefore, we are not working on it at this time. We plan to work on it when a stable version is released.

**Comment by vrozkovec** [comment]
Thank you for the information.

Honest question, how do you see that 1.20.4 is stable?

Checking here:
https://files.minecraftforge.net/net/minecraftforge/forge/index_1.20.4.html
https://forums.minecraftforge.net/topic/140300-forge-490-minecraft-1204/
https://maven.minecraftforge.net/net/minecraftforge/forge/1.20.4-49.0.14/forge-1.20.4-49.0.14-changelog.txt

it seems to me that 1.20.4 is stable.
If it is not too much to ask, would it be possible, please, to release a hotfix, in a separate branch, without releasing it to the curseforge?
I can compile and install the mod myself then. I'd try to fix the bug myself, but don't have the environment setup.

**Comment by noriokun4649** [comment]
1. Honest question, how do you see that 1.20.4 is stable?
`Recommended build` is stable version.  
Also minor version will be 1. ex: 49.1.0  

2. About crash log
This crash log is due to a bug in Forge.  
It seems that mods cannot be loaded if the file name contains "[” or "]".
But, well, even if I change the file name, I can't start it because it is not supported.
> ```
> Mod File: /home/vit/.minecraft/mods/[1.20][Forge]TorchBowMod_v1.0.jar
> Failure message: TorchBowMod has class loading errors
>     java.lang.IllegalArgumentException: Illegal character in path at index 33: file:///home/vit/.minecraft/mods/[1.20][Forge]TorchBowMod_v1.0.jar
> ```

3. Can you hotfix a separate branch?
yes.  
We have prepared branch 1.20.4 with some fixes.

**Comment by vrozkovec** [comment]
Wow, thank you, I see the change wasn't that trivial, thanks again, I'll try it out during the week.

**Comment by LexManos** [comment]
Correct, whenever paint gets around to the server run scripts. We can have a RB. There are no plans on any functional changes. The revival of the executable jar, and the performance improvements of no longer using the union file system for everything, which triggered this 'issue' is main functional changes that were intended for 1.20.4.

**Comment by noriokun4649** [comment]
@LexManos 
@PaintNinja 
Thank you for the valuable information provided by the maintainers and contributors.
I understand now that we do not plan to make any breaking changes.

@vrozkovec 
I have found that there are no plans to change the Forge features used in this mod.
Therefore, I will release a 1.20.4 version of TorchBowMod to CurseForge.

Also, the file name issue has been fixed in Forge 49.0.15.
